pkg/processor/build/runtime/java/runtime.go

```diff
@@ -0,0 +1,364 @@
+func (j *java) GetExtension() string {
+	return "jar"
+}
+
+// GetName returns the name of the runtime, including version if applicable
+func (j *java) GetName() string {
+	return "java"
+}
+
+// OnAfterStagingDirCreated will build jar if the source is a Java file
+// It will set generatedJarPath field
+func (j *java) OnAfterStagingDirCreated(stagingDir string) error {
+	buildPath := j.FunctionConfig.Spec.Build.Path
+	j.Logger.DebugWith(
+		"Java OnAfterStagingDirCreated", "buildPath", buildPath, "stagingDir", stagingDir)
+
+	handlerBuildDir := path.Join(stagingDir, "java-handler-build")
+	if err := os.Mkdir(handlerBuildDir, 0777); err != nil {
+		return errors.Wrap(err, "Can't create handler build dir")
+	}
+
+	userJarPath := path.Join(handlerBuildDir, userHandlerJarName)
+	if err := j.buildUserJar(buildPath, userJarPath); err != nil {
+		return err
+	}
+
+	var err error
+	handlerImageName := fmt.Sprintf("nuclio/handler-builder-java-onbuild:%s-%s", j.versionInfo.Label, j.versionInfo.Arch)
+	j.handlerJarPath, err = j.runDockerJavaBuild(handlerBuildDir, handlerImageName)
+
+	return err
+}
+
+func (j *java) getFunctionHandler() string {
+	return "Handler"
+}
+
+func (j *java) GetProcessorBaseImageName() (string, error) {
+	return fmt.Sprintf("nuclio/handler-java:%s-%s",
+		j.versionInfo.Label,
+		j.versionInfo.Arch), nil
+}
+
+// "reverser.jar:Reverser" -> "Reverser"
+func (j *java) handlerClassName(handler string) string {
+	fields := strings.Split(handler, ":")
+	if len(fields) == 1 {
+		return fields[0]
+	}
+	return fields[1]
+}
+
+func (j *java) createBuildFile(stagingBuildDir string) error {
+	buildFilePath := path.Join(stagingBuildDir, buildFileName)
+	if common.IsFile(buildFilePath) {
+		j.Logger.InfoWith("Found user build file, using it", "path", buildFilePath)
+		return nil
+	}
+
+	buildTemplate, err := template.New("build").Parse(buildTemplateCode)
+	if err != nil {
+		return err
+	}
+
+	buildFile, err := os.Create(buildFilePath)
+	if err != nil {
+		return err
+	}
+
+	defer buildFile.Close()
+
+	data := map[string]interface{}{
+		"Dependencies": j.FunctionConfig.Spec.Build.Dependencies,
+		"Handler":      j.handlerClassName(j.FunctionConfig.Spec.Handler),
+	}
+	return buildTemplate.Execute(buildFile, data)
+}
+
+func (j *java) createDockerFile(dockerfilePath, imageName string) error {
+	if !j.FunctionConfig.Spec.Build.NoBaseImagesPull {
+		// pull the onbuild image we need to build the processor builder
+		if err := j.DockerClient.PullImage(imageName); err != nil {
+			return errors.Wrap(err, "Failed to pull onbuild image for java")
+		}
+	}
+
+	dockerFileContent := fmt.Sprintf("FROM %s", imageName)
+
+	return ioutil.WriteFile(dockerfilePath, []byte(dockerFileContent), 0600)
+}
+
+func (j *java) isFile(filePath, extension string) bool {
+	return common.IsFile(filePath) && strings.ToLower(path.Ext(filePath)) == strings.ToLower(extension)
+}
+
+func (j *java) createJavaSourceDir() error {
```

Comment:
Looks like this function isn't used anywhere - remove?

Reply:
Removed.